### PR TITLE
FE-1548 Apply Validation HOCs to Textarea and Textbox based components

### PR DESCRIPTION
**Pull request by grabkowski**
- refactored validation HOCs to locate already existing validation issues,
- fixed an issue with validations not working outside of the Form component,
- added validation stories to Textbox and Date Input components,
- refactored Date Input to allow validation with new HOC,
- fixed an issue with InputIconToggle displayed next to the validation icon.

### CODE REVIEW

File: src/__experimental__/components/date-range/date-range.stories.js [hunk 1]
```diff
@@ -38,6 +38,8 @@ storiesOf('Experimental/Date Range', module)
   }, {
     notes: { markdown: notes },
     info: {
-      propTablesExclude: [State]
+      text: info,
+      propTablesExclude: [State],
+      notes: { markdown: notes }
```
Comment (izakrzewska) [inline src/__experimental__/components/date-range/date-range.stories.js]: double notes

File: src/__experimental__/components/textbox/textbox.stories.js [hunk 2]
```diff
@@ -16,17 +18,33 @@ const defaultStoryPropsConfig = {
...
         placeholder={ text('placeholder') }
         { ...getCommonTextboxStoryProps() }
       />
     );
+  }, {
+    info: {
+      text: info,
+      propTables: [OriginalTextbox],
+      propTablesExclude: [State, Textbox],
+      notes: { markdown: notes }
```
Comment (ianforster-sage) [inline src/__experimental__/components/textbox/textbox.stories.js]: I think textbox needs to handle special characters
`knobs: {
   escapeHTML: false
}`

File: src/__experimental__/components/textbox/textbox.stories.js [hunk 3]
```diff
@@ -40,6 +58,33 @@ storiesOf('Experimental/Textbox', module)
         { ...getCommonTextboxStoryProps() }
       />
     ]);
+  }, {
+    info: {
+      text: info,
+      propTables: [OriginalTextbox],
+      propTablesExclude: [State, Textbox],
+      notes: { markdown: notes }
```
Comment (ianforster-sage) [inline src/__experimental__/components/textbox/textbox.stories.js]: I think textbox needs to handle special characters
`knobs: {
   escapeHTML: false
}`

File: src/__experimental__/components/date/date.spec.js [hunk 4]
```diff
@@ -308,18 +349,46 @@ describe('Date', () => {
...
+      it('then these validations should be passed with internal validations to the Textbox Component', () => {
+        wrapper = render({ validations: [isNotFirstApr] });
+        const { internalValidations } = wrapper.instance().props;
+        expect(wrapper.find(Textbox).props().validations).toEqual([isNotFirstApr, ...internalValidations]);
+      });
+    });
+  });
 });
 
+function isNotFirstApr(value) {
```
Comment (izakrzewska) [inline src/__experimental__/components/date/date.spec.js]: is it possible to mock this function inside the test?

Comment (grabkowski) [reply src/__experimental__/components/date/date.spec.js]: you're right, function removed

**Comment by grabkowski** [comment]
> For Date Input I think you need to make a change to style JS file because the class `.icon-calendar` has property `cursor: pointer` but `.icon-error`, `.icon-warning` and `.icon-info` are using `cursor: text`, shouldn't they also use `cursor: pointer`?

good point, pointer added to Validation Styles

File: src/__experimental__/components/textbox/textbox.stories.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (grabkowski) [reply src/__experimental__/components/textbox/textbox.stories.js]: removed

Comment (grabkowski) [reply src/__experimental__/components/textbox/textbox.stories.js]: added

File: src/__experimental__/components/textbox/textbox.stories.js (hunk 2 above)
Comment (grabkowski) [reply src/__experimental__/components/textbox/textbox.stories.js]: added

File: src/__experimental__/components/date-range/date-range.stories.js (hunk 1 above)
Comment (grabkowski) [reply src/__experimental__/components/date-range/date-range.stories.js]: removed

**Pull request merged**